src/DefaultUnleash.php

```diff
@@ -152,6 +266,18 @@ private function isFeatureEnabled(?Feature $feature, Context $context, bool $def
             return new DefaultFeatureEnabledResult();
         }
 
+        $dependencies = $feature->getDependencies();
+        if (!empty($dependencies)) {
+            foreach ($dependencies as $dependency) {
+                $parentFeature = $parentFeatures[$dependency->getFeature()] ?? null;
+                if (!$this->isDependencySatisfied($dependency, $parentFeature, $context)) {
+                    $this->metricsHandler->handleMetrics($feature, false);
```

Comment:
this reminds me to add a test that verifies that metrics are not counted for the parent features